designs/osc_containerization/OSC-containerization.md

```diff
@@ -86,6 +88,49 @@ N/A
 ### OSC Synchronization Tasks
 N/A
 
+## OSC Kolla Container  
+In order for OSC to be available in the [TripleO docker repository](https://hub.docker.com/u/tripleoupstream/) it must build as part of the OpenStack Kolla.  The main requirement for this is the presence of a Docker file template `Dockerfile.j2` for OSC upstreamed to the [Kolla docker folder](https://github.com/openstack/kolla/tree/master/docker).  Below is the initial intended OSC Docker template file to be reviewed with the Kolla team:
+
+```
+FROM {{ namespace }}/{{ image_prefix }}base:{{ tag }}
+	
+LABEL maintainer="{{ maintainer }}" name="{{ image_name }}" build-date="{{ build_date }}"
+	
+{% block osc_header %}{% endblock %}
+{% import "macros.j2" as macros with context %}
+    
+{{ macros.configure_user(name='osc') }}
+	
+	
+# Copy OSC opt folder to container
+COPY . /
+EXPOSE 8090 443  # How to provide these parameters as part of the deployment?
+WORKDIR /opt/vmidc/bin/
+RUN chmod +x vmidc.sh 
+
+# Run OSC in console mode, otherwise container exists immediately
+CMD ["/opt/vmidc/bin/vmidc.sh", "--console", "--start"]
+```
+
+See details on how to [add a new service to Kolla here](https://docs.openstack.org/kolla/latest/contributor/CONTRIBUTING.html#adding-a-new-service).
+> Note: Upstream these changes to Kolla will likely require an [OpenStack blueprint](https://blueprints.launchpad.net/kolla).   
+
+> Questions for the Kolla team:
+> How different types of parameters can be passed to the container for launching?  
+> How to test the template locally? Do we need a Kolla deployment?  Can we have the OSC binaries on a local repository?  
+> What is the schedule to having the blueprint approved and what are the requirements?  
+> Other than the Jinja2 file above are there other files we must provide when upstreaming? For instance see PR https://review.openstack.org/#/c/416369/  
+> Are thee any specific storage restriction? How to ensure volumes are mounted correctly in the host when needed by the container?  
+> Is there anything specific for [ansible](https://github.com/openstack/kolla-ansible) or [kubernetes](https://github.com/openstack/kolla-kubernetes) we need to produce?  
+
+## OSC TripleO Service
+In order to the OSC service to be deployed as part of TripleO it must be registered as a service in the [TripleO service](designs/osc_containerization/OSC-containerization.md) for that changes to the tripleo-heat-templates must be done. The main configuration file needed will be a yaml file `services/osc-service.yaml`. This file will be similar to the [ODL template](designs/osc_containerization/OSC-containerization.md).
+
+> Questions for the Kolla/TripleO team:  
+> Which environment is needed to test the OSC service deployment with TripleO?  
+> Is a bluebrint needed for the changes in TripleO or can the Kolla blueprint above cover it all?  
+
```

Comment:
Can we also get the information on the time frame pertaining to:
1. acceptance of blueprint for a new feature like OSC
2. acceptance of OSC as a TripleO service.